src/diffusers/models/noise_augmentor.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from typing import List, Optional, Union
+
+import numpy as np
+import torch
+from torch import nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..utils import randn_tensor
+from .embeddings import Timesteps
+from .modeling_utils import ModelMixin
+
+
+# Copied from diffusers.schedulers.scheduling_ddpm.betas_for_alpha_bar
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999) -> torch.Tensor:
+    """
+    Create a beta schedule that discretizes the given alpha_t_bar function, which defines the cumulative product of
+    (1-beta) over time from t = [0,1].
+
+    Contains a function alpha_bar that takes an argument t and transforms it to the cumulative product of (1-beta) up
+    to that part of the diffusion process.
+
+
+    Args:
+        num_diffusion_timesteps (`int`): the number of betas to produce.
+        max_beta (`float`): the maximum beta to use; use values lower than 1 to
+                     prevent singularities.
+
+    Returns:
+        betas (`np.ndarray`): the betas used by the scheduler to step the model outputs
+    """
+
+    def alpha_bar(time_step):
+        return math.cos((time_step + 0.008) / 1.008 * math.pi / 2) ** 2
+
+    betas = []
+    for i in range(num_diffusion_timesteps):
+        t1 = i / num_diffusion_timesteps
+        t2 = (i + 1) / num_diffusion_timesteps
+        betas.append(min(1 - alpha_bar(t2) / alpha_bar(t1), max_beta))
+    return torch.tensor(betas, dtype=torch.float32)
+
+
+class NoiseAugmentor(ModelMixin, ConfigMixin):
```

Comment:
After some consideration I'm not really in favor of adding this class as is because:
- It doesn't fit into the current design - it's a mix of a model and a scheduler and we don't want scheduler logic under `src/diffusers/models`


For me there one of the following solutions is better:

1) If the mean and the variance vectors are learned during training, then it'd be better to create a `StableUnCLIPImageNormalizer` class that is added directly to the pipeline folder as is done here: https://github.com/huggingface/diffusers/blob/fd5c3c09afe29cbbde42cfa791ca598a07d94da9/src/diffusers/pipelines/unclip/text_proj.py#L22 and we potentially add a new scheduler (maybe called `noise_augmenter_scheduler` or `noise_aug_scheduler` which will just be a DDPM

2) If the mean and the variance vectors are not learned during training but fixed beforehand, then I don't think we need to add a `torch.nn.Module` class at all and we could just hardcode it into the pipeline for now and again add a new `noise_augmenter_scheduler` or `noise_aug_scheduler` scheduler. 
In this case, the `TimeSteps` class is not learned, no? So it could just be created in the pipeline's init.

Wdyt?

Reply:
Ah yes 100% to just making a separate scheduler for the noising process that makes a lot of sense.

I don't think we can hard code them in the pipeline because they're dependent on which CLIP is used but 100% to adding the small class that wraps the vectors.

Reply:
done here https://github.com/huggingface/diffusers/pull/2297/commits/ee5f35e33fbbc3aa4ecac20a9fc92f4a00d64418 !